src/app/content/practiceQuestions/hooks/locationChange.ts

```diff
@@ -22,11 +21,27 @@ const loadSummary = async(services: MiddlewareAPI & AppServices): Promise<Practi
 };
 
 const hookBody = (services: MiddlewareAPI & AppServices) => async() => {
-  const practiceQuestionsSummary = await loadSummary(services);
+  const state = services.getState();
```

Comment:
Hmm.. but should I move the first `hasPracticeQuestions` check into the `loadSummary` to remove one `if` check from `hookBody` or almost everything like it was before? But in this scenario - if `book` or `page` are `undefined`, it will throw error in `findArchiveTreeNodeById(book.tree, page.id);`. 

But moving the first `hasPracticeQuestions` into the `loadSummary` will make us use `getState()` again (3rd time).

Reply:
In the hookbody you should be able to have something like this:
```ts
  const practiceQuestionsSummary = await loadSummary(services);
  if (!practiceQuestionsSummary) { return; }
  
  services.dispatch(receivePracticeQuestionsSummary(practiceQuestionsSummary));
  
  const section = findArchiveTreeNodeById(book.tree, page.id);
  if (section && isLinkedArchiveTreeSection(section)) {
    services.dispatch(setSelectedSection(section));
  }
```